backend/simulation/realm/truck.py

```diff
@@ -37,13 +38,50 @@ def set_velocity(self, velocity: float) -> None:
 
     def reached_next_destination(self) -> None:
         self._route_index += 1
+        assert self._route_index <= len(self._route)
+        if self.done():
+            self._accumulated_reward += self.config.COMPLETION_REWARD
+            self._accumulated_info.append("Completed Journey")
 
     def done(self) -> bool:
         return self._route_index >= len(self._route)
 
     def set_current_truck_container(self, truck_container: TruckContainer) -> None:
         self._current_truck_container = truck_container
 
+    def compute_complete_route(self, realm: Realm) -> None:
+        i = 0
+        complete_route: List[int] = [self._current_truck_container.id_]
+        if complete_route[0] in realm.edges.keys():
+            complete_route[0] = realm.edges[complete_route[0]].start.id_
+        while i < len(self._route):
+            destination = self._route[i]
+            current = complete_route[-1]
+            if current == destination:
+                i += 1
+                continue
+            complete_route.append(realm.nodes[current].route_to(destination))
+        self._route = complete_route
+        self._route_index = 1
+
+    def collision(self, o: Truck) -> None:
+        # pylint: disable=unused-argument
+        self._accumulated_reward += self.config.COLLISION_REWARD
+        self._accumulated_info.append("Collision {o.id_}")
+
+    def tailgating(self, o: Truck) -> None:
+        # pylint: disable=unused-argument
+        self._accumulated_reward += self.config.TAILGATE_REWARD
+        self._accumulated_info.append("Tailgating {o.id_}")
```

Comment:
I assume these are meant to be f-strings (hence why pylint picks up on the unused-argument)